### PR TITLE
Modify src/shogun/classifier/ to implement CSignal in outer loops

### PR DESCRIPTION
**Pull request by abhinavrai44**
Issue #3508

### CODE REVIEW

File: src/shogun/classifier/AveragedPerceptron.cpp [hunk 1]
```diff
@@ -59,9 +60,11 @@ bool CAveragedPerceptron::train_machine(CFeatures* data)
 	for (int32_t i=0; i<num_feat; i++)
 		w[i]=1.0/num_feat;
 
+	CSignal::clear_cancel();
+
 	//loop till we either get everything classified right or reach max_iter
 
-	while (!converged && iter<max_iter)
+	while (!(CSignal::cancel_computations()) && (!converged && iter<max_iter))
 	{
```
Comment (karlnapf) [inline src/shogun/classifier/AveragedPerceptron.cpp]: I think we could also add an SG_INFO message for the iteration here if you want.....but seperate PR

**Comment by karlnapf** [comment]
Looks good, thanks for that. Ill merge as travis is fine
Did you try it from say python?

**Comment by abhinavrai44** [comment]
I tried the available python examples and they worked fine. One more thing some of the examples in examples/undocumented/libshogun seem to be outdated. For ex. classifier_newtontest.cpp has the include statement **#include <shogun/features/Labels.h>** but now Labels.h is the labels folder and also the functions associated with labels seem to have changed.

**Comment by abhinavrai44** [comment]
Also, the buildbot is giving error reports after this commit. Have I done something wrong ?

**Comment by karlnapf** [comment]
Yes, these examples are partly outdated, there is a blacklist in the `CMakeList.txt` that shows the examples that are not part of the test.

BTW feel free to port any of these examples to the new meta example systen, c.f. #3555

**Comment by karlnapf** [comment]
I dont think the buildbot error is your fault, but thanks for pointing it out!

File: src/shogun/classifier/AveragedPerceptron.cpp (hunk 1 above)
Comment (abhinavrai44) [inline src/shogun/classifier/AveragedPerceptron.cpp]: Yeah sure, but what should the message contain ? Which variables need to be highlighted..

Comment (karlnapf) [inline src/shogun/classifier/AveragedPerceptron.cpp]: Maybe just the iteration number.
If there is some residual that is checked to set the `converged` flag, you can also print that

Comment (abhinavrai44) [inline src/shogun/classifier/AveragedPerceptron.cpp]: ok

**Pull request merged**